controller/iteration.go

```diff
@@ -172,23 +196,35 @@ func (c *IterationController) Update(ctx *app.UpdateIterationContext) error {
 		return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
 	}
 
-	return application.Transactional(c.db, func(appl application.Application) error {
-		itr, err := appl.Iterations().Load(ctx.Context, id)
+	var itr *iteration.Iteration
+	var sp *space.Space
+	err = application.Transactional(c.db, func(appl application.Application) error {
+		itr, err = appl.Iterations().Load(ctx.Context, id)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, err)
+			return err
 		}
-		s, err := appl.Spaces().Load(ctx, itr.SpaceID)
+		sp, err = appl.Spaces().Load(ctx, itr.SpaceID)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, goa.ErrNotFound(err.Error()))
-		}
-		if !uuid.Equal(*currentUser, s.OwnerID) {
-			log.Warn(ctx, map[string]interface{}{
-				"space_id":     s.ID,
-				"space_owner":  s.OwnerID,
-				"current_user": *currentUser,
-			}, "user is not the space owner")
-			return jsonapi.JSONErrorResponse(ctx, errors.NewForbiddenError("user is not the space owner"))
+			return err
 		}
+		return nil
+	})
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, err)
+	}
+	authorized, spaceOwner, err := verifyUser(ctx, *currentUser, sp)
+	if err != nil {
+		return jsonapi.JSONErrorResponse(ctx, errors.NewUnauthorizedError(err.Error()))
+	}
+	if !authorized && !spaceOwner {
```

Comment:
If `!authorized` is `true`, we should return 401 instead of 403, right?

Reply:
please see https://github.com/fabric8-services/fabric8-wit/pull/1783/commits/f92e75707c4e9e632abb890011e820acb685271c